compiler/runtime/OMRRuntimeAssumptions.hpp

```diff
@@ -91,6 +101,24 @@ class RuntimeAssumption : public TR_Link0<RuntimeAssumption>
    uintptrj_t _key; // key for searching in the hashtable
    };
 
+class LocationRedirectRuntimeAssumption : public RuntimeAssumption
```

Comment:
If you're adding new classes, please add documentation describing what they're all about.  In particular, for the user, what does it mean to derive a class from one or the other?